add 2gig ct32 thermostat

chilicheech opened the pull request:
add 2gig ct32 thermostat, fix lint issues, and update readme thermostat example

robertsLando commented on README.md:
````diff
@@ -457,24 +457,24 @@ So in decimal format will become: `411-514-3`. This is the device id of `Heatit
 #### Thermostats
 
-```js
+```yaml
````
That's not yaml, it' JSON

chilicheech replied:
I know it's not `yaml`, but since `json` doesn't support comments and there's a comment in there, and  `json` is valid `yaml` I figured I would set it to `yaml` just so the syntax highlighting works properly with IDEs. I can change it to `json` and move the comments to outside the code block.

references:
https://stackoverflow.com/questions/21584985/what-valid-json-files-are-not-valid-yaml-1-1-files
https://en.wikipedia.org/wiki/YAML#Comparison_with_JSON

robertsLando commented on README.md:
````diff
@@ -497,24 +497,24 @@ Thermostats are most complex components to create, in this device example the se
 #### Fans
 
-```js
+```yaml
````
same

robertsLando commented on README.md:
````diff
@@ -533,18 +533,17 @@ Thermostats are most complex components to create, in this device example the se
 The main template is like the thermostat template. The things to add are:
 
-```js
+```yaml
````
same

robertsLando commented:
Everything is good here just change yaml with json and we can merge this. Thanks @chilicheech

robertsLando commented:
Yes you right that’s why I was using js as formatter (not json Sorry) 

Support me at:
    Github sponsors

> On 8 Apr 2020, at 18:16, Thiago Oliveira <notifications@github.com> wrote:
> 
> ﻿
> @chilicheech commented on this pull request.
> 
> In README.md:
> 
> > @@ -457,24 +457,24 @@ So in decimal format will become: `411-514-3`. This is the device id of `Heatit
>  
>  #### Thermostats
>  
> -```js
> +```yaml
> I know it's not yaml, but since json doesn't support comments and there's a comment in there, and json is valid yaml I figured I would set it to yaml just so the syntax highlighting works properly with IDEs. I can change it to json and move the comments to outside the code block.
> 
> references:
> https://stackoverflow.com/questions/21584985/what-valid-json-files-are-not-valid-yaml-1-1-files
> https://en.wikipedia.org/wiki/YAML#Comparison_with_JSON
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.

robertsLando commented on README.md:
````diff
@@ -533,61 +535,61 @@ Thermostats are most complex components to create, in this device example the se
-            "68-1-0", // <-- add fan values
-            "69-1-0"  // <-- add fan values
+##### GoControl GC-TBZ48 (Linear Nortek Security Control LLC):
+```json
````
```suggestion
```js
```

robertsLando commented on README.md:
````diff
@@ -497,24 +498,25 @@ Thermostats are most complex components to create, in this device example the se
-      state_value_template: "{% if (value_json.value | int) == 0 %} 0 {% else %} 99 {% endif %}",
-      speed_value_template: "{% if (value_json.value | int) == 25 %}  24  {% elif (value_json.value | int) == 51 %} 50 {% elif (value_json.value | int) == 99 %} 99 {% else %}  0  {% endif %}"
+##### GE 1724 Dimmer:
+```json
````
```suggestion
```js
```

robertsLando commented on README.md:
````diff
@@ -457,24 +457,25 @@ So in decimal format will become: `411-514-3`. This is the device id of `Heatit
-      temperature_state_template: '{{ value_json.value }}',
-      temperature_command_topic: true
+##### Heatit Thermostat TF 021 (ThermoFloor AS):
+```json
````
```suggestion
```js
```

chilicheech commented:
Ok, so the reason I changed it to json is because of `customDevices.json`. It seems simpler to use that instead of `customDevices.js`, especially because it support auto reloading.

I was looking at the rendered output in GitHub and I think the syntax highlighting looks better with yaml instead of json. With yaml the keys have a different color than the values. With json it's all the same color. And since json is valid yaml it made sense to me. Folks reading the readme won't know that it says yaml, it just helps the application render the visualization and checks correctly.

Also, by switching to `js` my IDE still complains, whereas with `yaml` it doesn't:
js:
![g-js](https://user-images.githubusercontent.com/732659/78918569-786d9100-7a45-11ea-872f-208b5b7ce056.png)
yaml:
![g-yaml](https://user-images.githubusercontent.com/732659/78918585-7c99ae80-7a45-11ea-9afa-7ef474ca532a.png)

I changed it back to `js`. If you want I can change the example back to js from json too.

Cheers,

robertsLando commented:
Thanks @chilicheech !
